### PR TITLE
[WIP] Modifications to work with SDL2 on Android

### PR DESCRIPTION
**Pull request by inclement**
These are actually quite minimal, and are all that is actually needed to work with SDL2. Some of the details will need to be updated later (e.g. changing the NewPythonActivity references, and maybe changing how Hardware.java works more than I already did), but I believe these can and should be merged now, it will make it easier to test the sdl2 build tools.

Edit: Actually, not to be merged yet.

### CODE REVIEW

**Comment by inclement** [comment]
In one place I left the old android module call for use with pygame, but added the sdl2 one with pyjnius. This is because I had to modify Hardware.java with the line at https://github.com/inclement/p4a-experiment/blob/master/bootstrap_templates/sdl2/src/org/renpy/android/Hardware.java#L166 in order for it to work...I haven't tested if this is necessary with pygame (is it done somewhere else?).

File: kivy/metrics.py [hunk 1]
```diff
@@ -158,8 +159,13 @@ def dpi(self):
             return float(custom_dpi)
 
         if platform == 'android':
-            import android
-            return android.get_dpi()
+            if USE_SDL2:
```
Comment (tito) [inline kivy/metrics.py]: i think we could always use jnius now.

File: kivy/metrics.py [hunk 2]
```diff
@@ -217,8 +223,13 @@ def fontscale(self):
 
         if platform == 'android':
             from jnius import autoclass
-            PythonActivity = autoclass('org.renpy.android.PythonActivity')
-            config = PythonActivity.mActivity.getResources().getConfiguration()
+            if USE_SDL2:
+                # This activity name is temporary, it will need to be changed later
+                NewPythonActivity = autoclass('net.inclem.android.NewPythonActivity')
```
Comment (tito) [inline kivy/metrics.py]: org.kivy ? :)

Comment (inclement) [inline kivy/metrics.py]: Yep, I'll definitely change it later, I started this way to be clear it wasn't final :)

File: kivy/metrics.py (hunk 1 above)
Comment (inclement) [inline kivy/metrics.py]: I had to modify Hardware.java to make it work (maybe due to an android bug, http://stackoverflow.com/questions/27590213/android-displaymetrics-densitydpi-returns-zero-on-rooted-devices) and I didn't test that with the pygame bootstrap. I'll change it all to pyjnius if that works.

**Comment by inclement** [comment]
Replaced by #3648

**Pull request closed without merging**